AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -333,16 +349,26 @@ - (void)setAsyncDataSource:(id<ASCollectionViewDataSource>)asyncDataSource
   if (asyncDataSource == nil) {
     _asyncDataSource = nil;
     _proxyDataSource = _isDeallocating ? nil : [[ASCollectionViewProxy alloc] initWithTarget:nil interceptor:self];
-    _asyncDataSourceImplementsConstrainedSizeForNode = NO;
-    _asyncDataSourceImplementsNodeBlockForItemAtIndexPath = NO;
+    
+    _asyncDataSourceFlags.asyncDataSourceConstrainedSizeForNode = NO;
```

Comment:
It would be great to take advantage of the fact that these have been cleanly stored in a single flags variable, and find a way to zero all of them together. Ideally, this would work even if a new flag is added later, Maybe by making use of sizeof().  Not exactly sure how to do this.  Would setting _asyncDataSourceFlags = 0ULL do the trick?